### PR TITLE
[feature](docker suite) make docker suite no need external doris cluster

### PR DESCRIPTION
**Pull request by yujun777**
1.  Make docker suite no need external doris cluster:
    a. run docker suite according to ClusterOptions.cloudMode = true/false;
    b. if ClusterOptions.cloudMode = null,   user can set it with cmd:    `sh run-regression-test.sh  --run docker_action  -runMode=cloud/not_cloud`

2. refactor database.py, simplifier its logic;

3.  doris compose add cmd options -v for debug log.

4.  doris compose print flush buffer.

### CODE REVIEW

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

File: regression-test/framework/src/main/groovy/org/apache/doris/regression/suite/Suite.groovy [hunk 1]
```diff
@@ -283,15 +284,25 @@ class Suite implements GroovyInterceptable {
...
-            dockerIsCloud = pipelineIsCloud
+            if (context.config.runMode == RunMode.UNKNOWN) {
+                throw new Exception("Bad run mode, cloud or not_cloud is unknown")
+            }
+            dockerIsCloud = context.config.runMode == RunMode.CLOUD
         } else {
-            dockerIsCloud = options.cloudMode
-            if (dockerIsCloud != pipelineIsCloud && options.skipRunWhenPipelineDiff) {
+            if (options.cloudMode == true && context.config.runMode == RunMode.NOT_CLOUD) {
                 return
```
Comment (deardeng) [inline regression-test/framework/src/main/groovy/org/apache/doris/regression/suite/Suite.groovy:300]: add log

File: regression-test/framework/src/main/groovy/org/apache/doris/regression/suite/Suite.groovy [hunk 2]
```diff
@@ -283,15 +284,25 @@ class Suite implements GroovyInterceptable {
...
             }
+            if (options.cloudMode == false && context.config.runMode == RunMode.CLOUD) {
+                return
```
Comment (deardeng) [inline regression-test/framework/src/main/groovy/org/apache/doris/regression/suite/Suite.groovy:303]: add log

**Review by deardeng — APPROVED** [review]
LGTM

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run p0

**Comment by yujun777** [comment]
run p0

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run cloud_p0

**Comment by yujun777** [comment]
run cloud_p0

**Comment by yujun777** [comment]
run cloud_p0

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**